### PR TITLE
[SDPA-4419] update content moderation filters

### PR DESCRIPTION
**Pull request by asarfara**
**JIRA issue:** https://digital-engagement.atlassian.net/browse/SDPA-4419

**Added**
Added filters i.e. primary site and subsite for moderation content screen 

**Removed**
Removed language filter from moderation content screen 
Removed moderation state filter from overview content screen

**Related PR**
https://github.com/dpc-sdp/content-vic-gov-au/pull/897

### CODE REVIEW

File: config/install/field.storage.node.field_node_primary_site.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+type: entity_reference
+settings:
+  target_type: taxonomy_term
+module: core
+locked: false
+cardinality: 1
+translatable: true
+indexes: {  }
+persist_with_no_fields: false
+custom_storage: false
```
Comment (vincent-gao) [inline config/install/field.storage.node.field_node_primary_site.yml:19]: hey @asarfara , tide_core doesn't have dependency with tide_site. so this file doesn't need to be added.

File: config/install/field.storage.node.field_node_site.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+type: entity_reference
+settings:
+  target_type: taxonomy_term
+module: core
+locked: false
+cardinality: -1
+translatable: true
+indexes: {  }
+persist_with_no_fields: false
+custom_storage: false
```
Comment (vincent-gao) [inline config/install/field.storage.node.field_node_site.yml:19]: no need this file as the same reason described above

File: config/install/taxonomy.vocabulary.sites.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+langcode: en
+status: true
+dependencies: {  }
+name: Sites
+vid: sites
+description: 'Sites and sections'
+hierarchy: 0
+weight: 0
```
Comment (vincent-gao) [inline config/install/taxonomy.vocabulary.sites.yml:8]: no need this file as the same reason described above

File: config/install/views.view.tide_core_moderated_content_filters.yml [hunk 4]
```diff
@@ -2,10 +2,14 @@ langcode: en
...
 dependencies:
   config:
+    - field.storage.node.field_node_primary_site
+    - field.storage.node.field_node_site
+    - taxonomy.vocabulary.sites
     - workflows.workflow.editorial
   module:
     - content_moderation
     - node
+    - taxonomy
```
Comment (vincent-gao) [inline config/install/views.view.tide_core_moderated_content_filters.yml:12]: no need this line

File: config/install/views.view.tide_core_moderated_content_filters.yml [hunk 5]
```diff
@@ -2,10 +2,14 @@ langcode: en
 status: true
 dependencies:
   config:
+    - field.storage.node.field_node_primary_site
+    - field.storage.node.field_node_site
+    - taxonomy.vocabulary.sites
```
Comment (vincent-gao) [inline config/install/views.view.tide_core_moderated_content_filters.yml:7]: the three lines above are not required.

File: tide_core.install [hunk 6]
```diff
@@ -672,3 +672,36 @@ function tide_core_update_8025() {
...
+
+  $config->set('display.default.cache_metadata.tags', [
+    'config:field.storage.node.field_node_primary_site',
+    'config:field.storage.node.field_node_site',
+  ]);
+
+  $config->set('display.page_1.cache_metadata.tags', [
+    'config:field.storage.node.field_node_primary_site',
+    'config:field.storage.node.field_node_site',
+  ]);
```
Comment (vincent-gao) [inline tide_core.install:704]: hey @asarfara . no need line 696 to line 704

File: config/install/views.view.tide_core_moderated_content_filters.yml [hunk 7]
```diff
@@ -192,6 +196,132 @@ display:
           entity_type: node
           entity_field: title
           plugin_id: field
+        field_node_primary_site:
```
Comment (vincent-gao) [inline config/install/views.view.tide_core_moderated_content_filters.yml:199]: hey @asarfara ,I think we might want to delete `moderation_state` filter only from the yaml file.

**Comment by vincent-gao** [comment]
hey @asarfara , one more little thing,  the title,`Feature/sdpa 4419 update content moderation filters` ,should be changed to `[SDPA-4419] update content moderation filters`

File: tide_core.install [hunk 8]
```diff
@@ -672,3 +672,36 @@ function tide_core_update_8025() {
     ->set('register', UserInterface::REGISTER_VISITORS_ADMINISTRATIVE_APPROVAL)
     ->save(TRUE);
 }
+
+/**
+ * Import tide_core_moderated_content_filters.
+ */
+function tide_core_update_8026() {
```
Comment (vincent-gao) [inline tide_core.install:679]: I think this hook is unnecessary

File: config/install/views.view.tide_core_moderated_content_filters.yml (hunk 7 above)
Comment (asarfara) [reply config/install/views.view.tide_core_moderated_content_filters.yml:199]: Hi @vincent-gao this is moderation content view screen. This screen is different to overview content view screen.

Primary site and site filters needs to be added to Content moderation view.
Moderation state needs to be removed from Overview content view.

File: tide_core.install [hunk 9]
```diff
@@ -672,3 +672,36 @@ function tide_core_update_8025() {
...
+}
+
+/**
+ * Update content overview filters.
+ */
+function tide_core_update_8027() {
+  $config_factory = \Drupal::configFactory();
+  $config = $config_factory->getEditable('views.view.summary_contents_filters');
+
+  $config->clear('display.default.display_options.filters.moderation_state');
```
Comment (asarfara) [inline tide_core.install:694]: This needs to be moved to tide_site.

**Pull request closed without merging**